micro-rdk/src/common/data_collector.rs

```diff
@@ -0,0 +1,352 @@
+use std::fmt::Display;
+
+use crate::proto::app::data_sync::v1::SensorData;
+
+use super::{
+    config::{AttributeError, Kind},
+    movement_sensor::MovementSensor,
+    robot::ResourceType,
+    sensor::{get_sensor_readings_data, SensorError},
+};
+
+use thiserror::Error;
+
+/// A DataCollectorConfig instance is a representation of an element
+/// of the list of "capture_methods" in the "attributes" section of a
+/// component's configuration JSON object as stored in app. Each element
+/// of "capture_methods" is meant to produce an instance of `DataCollector`
+/// as defined below
+#[derive(Debug, Clone)]
+pub struct DataCollectorConfig {
+    pub method: CollectionMethod,
+    pub capture_frequency_hz: f32,
+}
+
+impl TryFrom<&Kind> for DataCollectorConfig {
+    type Error = AttributeError;
+    fn try_from(value: &Kind) -> Result<Self, Self::Error> {
+        if !value.contains_key("method")? {
+            return Err(AttributeError::KeyNotFound("method".to_string()));
+        }
+        if !value.contains_key("capture_frequency_hz")? {
+            return Err(AttributeError::KeyNotFound(
+                "capture_frequency_hz".to_string(),
+            ));
+        }
+        let method_str: String = value
+            .get("method")?
+            .ok_or(AttributeError::KeyNotFound("method".to_string()))?
+            .try_into()?;
+        let capture_frequency_hz = value
+            .get("capture_frequency_hz")?
+            .ok_or(AttributeError::KeyNotFound(
+                "capture_frequency_hz".to_string(),
+            ))?
+            .try_into()?;
+        // TODO: Collectors that take arguments (ex. Board Analogs)
+        // let parameters: &Kind = if let Ok(Some(params)) = value.get("additional_params") {
+        //     params
+        // } else {
+        //     &Kind::NullValue(0)
+        // };
```

Comment:
I think maybe remove the commented out code under the `TODO` sections (here and elsewhere) in favor of just a description. The code is unlikely to remain useful and correct because as things change under review these probably won't keep up.

Reply:
sorry I think this was a miscommunication on my part, but it's my plan to delete the TODO comments before merge (although on reflection, I like keeping the comment and removing the commented-out code, thoughts?)

Reply:
Keep the TODOs, with as much explication as you see fit, but drop the code would be my recommendation.